moha/hamiltonians.py

```diff
@@ -421,30 +425,67 @@ def generate_one_body_integral(self,
         -------
         scipy.sparse.csr_matrix or np.ndarray
         """
-        if basis != 'spin orbital':
+        if basis != 'spinorbital basis':
             raise ValueError('Selected Hamiltonian supports'
-                             ' only spin orbital basis')
-        one_body_term = 0.5*diags(self.mu - np.diag(self.J_eq) -
-                                  np.sum(self.J_ax, axis=1),
-                                  format="csr")
+                             ' only spinorbital basis')
+        one_body_term = 0.5 * diags(self.mu - np.diag(self.J_eq) -
+                                    np.sum(self.J_ax, axis=1),
+                                    format="csr")
         self.one_body = one_body_term
         return self.one_body.todense() if dense else self.one_body
 
     def generate_two_body_integral(self,
                                    sym: int,
                                    dense: bool,
-                                   basis='spinorbital'):
-        """
-        Generate two body integral.
+                                   basis='spinorbital basis'):
+        r"""
+        Generate two body integral in spatial or spinorbital basis.
 
         Parameters
         ----------
-        sym
-        dense
-        basis
+        basis: str
+            ['spin orbital']
+        dense: bool
+            dense or sparse matrix; default False
+        sym: int
+            symmetry -- [2, 4, 8] default is 1
 
         Returns
         -------
-        None
+        scipy.sparse.csr_matrix or np.ndarray
         """
-        pass
+        if basis != 'spinorbital basis':
+            raise ValueError('Selected Hamiltonian supports'
+                             ' only spinorbital basis')
+
+        n_sp = self.n_sites//2
```

Comment:
Taking n_sp to refer to the number of spatial orbitals,
I think this should be:
n_sp = self.n_sites

and if v is the 2-body integrals in spin-orbital basis, we should have:
v = lil_matrix((2*n_sites * 2*n_sites, 2*n_sites * 2*n_sites))
much like what was defined for PPP Hamiltonian

Reply:
It's OK to do it in the spatial basis I think. Did we write a general spin-orbital to spatial-orbital transform? If so it should work here. The only case (that I'm aware of) where spin-basis exists and spatial-basis doesn't is where there is a (real) magnetic field, which couples with the (by convention) alpha- and beta-spin orbitals in differnet ways.

While Heisenberg is a spin model, once we transform it to the 2nd-quantized representation (each spin = 1 spatial orbital, as @gabrielasd notes) then it's spatial-basis-representable.